src/Api/CMakeLists.txt

```diff
@@ -14,15 +14,29 @@ target_sources(ApiInterface INTERFACE
 target_include_directories(ApiInterface INTERFACE
         ${CMAKE_CURRENT_LIST_DIR}/include)
 
+
+add_library(ApiBase STATIC)
+target_compile_options(ApiBase PRIVATE ${STRICT_COMPILE_FLAGS})
+target_compile_features(ApiBase PUBLIC cxx_std_17)
+
+target_sources(ApiBase PUBLIC
+        include/Api/EncodedString.h)
+target_sources(ApiBase PRIVATE
+        EncodedString.cpp)
+target_link_libraries(ApiBase PUBLIC
+        GrpcProtos)
+
 add_executable(ApiInterfaceTests)
 
 target_compile_options(ApiInterfaceTests PRIVATE ${STRICT_COMPILE_FLAGS})
 
 target_sources(ApiInterfaceTests PRIVATE
-        EncodedEventTest.cpp)
+        EncodedEventTest.cpp
+        EncodedStringTest.cpp)
 
 target_link_libraries(ApiInterfaceTests PRIVATE
         ApiInterface
+        ApiBase
```

Comment:
Do you need the new `ApiBase` target then?

Reply:
In my understanding yes. "Api" is the shared library that we will be injected to the target. We should (and do not) depend on that in other Orbit modules. But "ApiInterface" is just an interface, and can not include private implementation files. I could make it Header only, but that also feels wrong.

Reply:
Is there a specific reason why "ApiInterface" in just an interface? Or was it simply because until now it only included header files and hence had to be an interface?

Reply:
I think otherwise, it would complain about the missing implementation of Orbit.h.